extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -127,6 +127,21 @@ export function isPositionFixed(el, win) {
   return hasFixedAncestor;
 }
 
+/**
+ * @param {!Element} el
+ * @param {!Window} win
+ * @return {string|null} a string of the name of valid containers of the ad.
+ * This is called before creating iframe for this ad.
+ */
+export function getAdContainer(el) {
+  do {
```

Comment:
This is needed because for `flying-carpet` allows wrapping ad in other element. 
For example <amp-fx-flying-carpet><div><amp-ad></amp-ad></div></amp-fx-flying-carpet> is valid.